ruby/ql/lib/codeql/ruby/frameworks/core/String.qll

```diff
@@ -151,7 +131,15 @@ module String {
     }
   }
 
-  // TODO: we already have a summary for Array#clear. Check that it applies correctly to String#clear.
+  /**
+   * This is a placeholder for `String#clear`.
+   * We can't currently write this summary because there is no `DataFlow::Content` node to refer to (unlike with `Array#clear`).
+   * We need a `DataFlow::Content` node in order to override `clearsContent`.
+   */
+  private class ClearSummary extends SimpleSummarizedCallable {
```

Comment:
We don't need to do it on this PR, but I believe some other languages have defined default barriers, and I guess this would be a good candidate for that.

Reply:
That seems like a good idea!